src/runner/basic.rs

```diff
@@ -590,22 +718,87 @@ async fn execute<W, Before, After>(
             }
         }
 
-        while let Ok(Some((feat, rule, scenario_failed))) =
+        while let Ok(Some((feature, rule, scenario, scenario_failed))) =
             storage.finished_receiver.try_next()
         {
-            if let Some(r) = rule {
+            if let Some(r) = rule.clone() {
                 if let Some(f) =
-                    storage.rule_scenario_finished(Arc::clone(&feat), r)
+                    storage.rule_scenario_finished(feature.clone(), r)
                 {
                     executor.send_event(f);
                 }
             }
-            if let Some(f) = storage.feature_scenario_finished(feat) {
+            if let Some(f) = storage.feature_scenario_finished(feature.clone())
+            {
                 executor.send_event(f);
             }
 
-            if fail_fast && scenario_failed {
-                started_scenarios = ControlFlow::Break(());
+            if scenario_failed {
+                if fail_fast {
+                    started_scenarios = ControlFlow::Break(());
+                } else {
+                    // For some reason features.scenarios contains an empty vector for the storage.
+                    // It will fail the rest of the execution. How to access the storage so we can
+                    // read and update the retries?
```

Comment:
@ilslv can you maybe help me to understand this part? I try to read and update the storage. Seems like it's empty at this step (the scenario type is set to `Concurrent` but there is no vector as a value).